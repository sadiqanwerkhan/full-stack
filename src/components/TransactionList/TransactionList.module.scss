@import '../../styles/variables.scss';

.transactionList {
    background-color: $white;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    border: 1px solid $n300;
    border-radius: $spacing-mega;
    padding: 0px;
    margin-bottom: $spacing-kilo;
  
    li {
      align-items: center;
      position: relative;
      border-bottom: 1px solid $n300;
      padding: $spacing-giga;
  
      &:first-of-type {
        border-top-left-radius: $spacing-mega;
        border-top-right-radius: $spacing-mega;
      }
  
      &:last-of-type {
        border-bottom-left-radius: $spacing-mega;
        border-bottom-right-radius: $spacing-mega;
        border-bottom: 0;
      }
    }
  }