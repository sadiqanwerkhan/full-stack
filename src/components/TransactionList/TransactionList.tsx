import React, { FC, PropsWithChildren, memo } from "react";

import styles from "./TransactionList.module.scss";

const TransactionList = memo(({ children }: PropsWithChildren) => {
  return <ul className={styles.transactionList}>{children}</ul>;
});

export { TransactionList };
