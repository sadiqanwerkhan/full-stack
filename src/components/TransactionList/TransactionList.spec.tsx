import React from "react";

import { render } from "@testing-library/react";

import { TransactionList } from "./TransactionList";

describe("TransactionList", () => {
  it("should render matching default styles", () => {
    const { asFragment } = render(
      <TransactionList>
        <li>1</li>
        <li>2</li>
        <li>3</li>
      </TransactionList>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
