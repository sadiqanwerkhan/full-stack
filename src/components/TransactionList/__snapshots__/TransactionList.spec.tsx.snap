// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList should render matching default styles 1`] = `
<DocumentFragment>
  <ul
    class="transactionList"
  >
    <li>
      1
    </li>
    <li>
      2
    </li>
    <li>
      3
    </li>
  </ul>
</DocumentFragment>
`;
